src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs

```diff
@@ -421,6 +422,454 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
         Assert.True(label.IsHandleCreated);
     }
 
+    [WinFormsFact]
+    public void Label_AutoSizeChangedEvent_AddRemove_Success()
+    {
+        using Label label = new();
+        int callCount = 0;
+        EventHandler handler = (sender, e) =>
+        {
+            Assert.Same(label, sender);
+            Assert.Same(EventArgs.Empty, e);
+            callCount++;
+        };
+
+        label.AutoSizeChanged += handler;
+        label.AutoSizeChanged -= handler;
+        Assert.Equal(0, callCount);
+
+        label.AutoSizeChanged += handler;
+        label.AutoSize = label.AutoSize;
+        Assert.Equal(0, callCount);
+
+        label.AutoSize = !label.AutoSize;
+        Assert.Equal(1, callCount);
+
+        label.AutoSize = label.AutoSize;
+        Assert.Equal(1, callCount);
+
+        label.AutoSize = !label.AutoSize;
+        Assert.Equal(2, callCount);
+
+        label.AutoSizeChanged -= handler;
+        label.AutoSize = !label.AutoSize;
+        Assert.Equal(2, callCount);
+    }
+
+    [WinFormsFact]
+    public void Label_AutoEllipsis_SetWithHandle_GetReturnsExpected()
+    {
+        using Label label = new();
+        Assert.NotEqual(0, label.Handle);
+
+        int invalidatedCallCount = 0;
+        label.Invalidated += (sender, e) => invalidatedCallCount++;
+        int styleChangedCallCount = 0;
+        label.StyleChanged += (sender, e) => styleChangedCallCount++;
+        int createdCallCount = 0;
+        label.HandleCreated += (sender, e) => createdCallCount++;
+
+        // Set true.
+        label.AutoEllipsis = true;
+        Assert.True(label.AutoEllipsis);
+        Assert.True(label.IsHandleCreated);
+        Assert.Equal(1, invalidatedCallCount);
+        Assert.Equal(0, styleChangedCallCount);
+        Assert.Equal(0, createdCallCount);
+
+        // Set same.
+        label.AutoEllipsis = true;
+        Assert.True(label.AutoEllipsis);
+        Assert.True(label.IsHandleCreated);
+        Assert.Equal(1, invalidatedCallCount);
+        Assert.Equal(0, styleChangedCallCount);
+        Assert.Equal(0, createdCallCount);
+
+        // Set different.
+        label.AutoEllipsis = false;
+        Assert.False(label.AutoEllipsis);
+        Assert.True(label.IsHandleCreated);
+        Assert.Equal(2, invalidatedCallCount);
+        Assert.Equal(0, styleChangedCallCount);
+        Assert.Equal(0, createdCallCount);
+    }
+
+    [WinFormsFact]
+    public void Label_BackgroundImageGetSet()
+    {
+        using Label label = new();
+        Assert.Null(label.BackgroundImage);  // Default value
+
+        // Set image.
+        using Bitmap image1 = new Bitmap(10, 10);
```

Comment:
Please use target type new, ditto for some other tests below as well.
```suggestion
        using Bitmap image1 = new(10, 10);
```